crates/core/thin_edge_json/src/event.rs

```diff
@@ -39,27 +42,47 @@ pub mod error {
 
         #[error(transparent)]
         SerdeJsonError(#[from] serde_json::error::Error),
+
+        #[error(transparent)]
+        JsonWriterError(#[from] JsonWriterError),
     }
 }
 
 impl ThinEdgeEvent {
     pub fn try_from(
         mqtt_topic: &str,
         mqtt_payload: &str,
+        may_be_child: Option<&str>,
     ) -> Result<Self, ThinEdgeJsonDeserializerError> {
```

Comment:
Adding this `may_be_child` parameter is really a bad idea. This child id is extracted from the second argument i.e the topic name.

This makes the code more complex with:
* two places where the topic name is decoded,
* possible inconsistency when the two child device ids (given by the first and third arguments) are different.

Reply:
refactored.